dbms/src/Functions/FunctionsIsIPAddr.h

```diff
@@ -0,0 +1,142 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <Columns/ColumnString.h>
+#include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypesNumber.h>
+#include <Functions/FunctionHelpers.h>
+#include <Functions/IFunction.h>
+#include <arpa/inet.h>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+} // namespace ErrorCodes
+
+/** Helper functions
+  *
+  * isIPv4(x) - Judge whether the input string is an IPv4 address.
+  *
+  * isIPv6(x) - Judge whether the input string is an IPv6 address
+  *
+  */
+
+static inline UInt8 isIPv4(String input_address)
+{
+    if (input_address.empty())
+        return 0;
+    char str[INET_ADDRSTRLEN];
+    if (inet_pton(AF_INET, input_address.c_str(), str) == 1)
```

Comment:
```
`inet_pton` converts the character string src into a network
       address structure in the af address family, then copies the
       network address structure to dst.
```

http://svn.apache.org/repos/asf/apr/apr/trunk/network_io/unix/inet_pton.c

What we need to do is just check whether the input string is an IP address. So I found there are unnecessary memory copy, bounds checking and length counting in `inet_pton.c`, can we copy the code and optimize it? But It is also fine for me to use `inet_pton` directly.

Reply:
done, Please take another look, thank you very much.